src/utringbuffer.h

```diff
@@ -94,7 +94,7 @@ typedef struct {
 
 #define _utringbuffer_real_idx(a,j) ((a)->f ? ((j) + (a)->i) % (a)->n : (j))
 #define _utringbuffer_internalptr(a,j) ((void*)((a)->d + ((a)->icd.sz * (j))))
-#define utringbuffer_eltptr(a,j) ((0 <= (j) && (j) < utringbuffer_len(a)) ? _utringbuffer_internalptr(a,_utringbuffer_real_idx(a,j)) : NULL)
+#define utringbuffer_eltptr(a,j) (((j) < utringbuffer_len(a)) ? _utringbuffer_internalptr(a,_utringbuffer_real_idx(a,j)) : NULL)
```

Comment:
Here `j` could certainly be negative if it were, for example, `-1`, or an `int` that happened to have the value `-1`.
If you're doing this to work around a compiler warning, please file a bug/issue with a minimal compilable example that demonstrates the warning and maybe we can find a way to silence it.

Reply:
Hi, here's the output from the compiler and seem to disagree with that assertion but it might be wrong. 

```In file included from swuniq.c:41:
swuniq.c: In function ‘lookup’:
uthash/src/utringbuffer.h:97:38: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
   97 | #define utringbuffer_eltptr(a,j) ((0 <= (j) && (j) < utringbuffer_len(a)) ? _utringbuffer_internalptr(a,_utringbuffer_real_idx(a,j)) : NULL)
      |                                      ^~
swuniq.c:133:25: note: in expansion of macro ‘utringbuffer_eltptr’
  133 |    XXH64_hash_t *item = utringbuffer_eltptr(rbuffer, i);
      |                         ^~~~~~~~~~~~~~~~~~~
swuniq.c: In function ‘main’:
uthash/src/utringbuffer.h:46:12: warning: conversion from ‘size_t’ {aka ‘long unsigned int’} to ‘unsigned int’ may change value [-Wconversion]
   46 |   (a)->n = (_n);                                                          \
      |            ^
uthash/src/utringbuffer.h:76:3: note: in expansion of macro ‘utringbuffer_init’
   76 |   utringbuffer_init(a, n, _icd);                                          \
      |   ^~~~~~~~~~~~~~~~~
swuniq.c:176:2: note: in expansion of macro ‘utringbuffer_new’
  176 |  utringbuffer_new(history, window_size, &ut_xxh64_hash_t_icd);
      |  ^~~~~~~~~~~~~~~~
uthash/src/utringbuffer.h:97:38: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
   97 | #define utringbuffer_eltptr(a,j) ((0 <= (j) && (j) < utringbuffer_len(a)) ? _utringbuffer_internalptr(a,_utringbuffer_real_idx(a,j)) : NULL)
      |                                      ^~
uthash/src/utringbuffer.h:55:23: note: in expansion of macro ‘utringbuffer_eltptr’
   55 |         (a)->icd.dtor(utringbuffer_eltptr(a, _ut_i));                     \
      |                       ^~~~~~~~~~~~~~~~~~~
uthash/src/utringbuffer.h:69:3: note: in expansion of macro ‘utringbuffer_clear’
   69 |   utringbuffer_clear(a);                                                  \
      |   ^~~~~~~~~~~~~~~~~~
uthash/src/utringbuffer.h:80:3: note: in expansion of macro ‘utringbuffer_done’
   80 |   utringbuffer_done(a);                                                   \
      |   ^~~~~~~~~~~~~~~~~
swuniq.c:189:2: note: in expansion of macro ‘utringbuffer_free’
  189 |  utringbuffer_free(history);
      |  ^~~~~~~~~~~~~~~~~
uthash/src/utringbuffer.h:97:38: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
   97 | #define utringbuffer_eltptr(a,j) ((0 <= (j) && (j) < utringbuffer_len(a)) ? _utringbuffer_internalptr(a,_utringbuffer_real_idx(a,j)) : NULL)
      |                                      ^~
uthash/src/utringbuffer.h:60:23: note: in expansion of macro ‘utringbuffer_eltptr’
   60 |         (a)->icd.dtor(utringbuffer_eltptr(a, _ut_i));                     \
      |                       ^~~~~~~~~~~~~~~~~~~
uthash/src/utringbuffer.h:69:3: note: in expansion of macro ‘utringbuffer_clear’
   69 |   utringbuffer_clear(a);                                                  \
      |   ^~~~~~~~~~~~~~~~~~
uthash/src/utringbuffer.h:80:3: note: in expansion of macro ‘utringbuffer_done’
   80 |   utringbuffer_done(a);                                                   \
      |   ^~~~~~~~~~~~~~~~~
swuniq.c:189:2: note: in expansion of macro ‘utringbuffer_free’
  189 |  utringbuffer_free(history);
      |  ^~~~~~~~~~~~~~~~~
```